### ISSUE TITLE
Create new model structure for BurtCo

### ISSUE THREAD

**Issue by kmcelwee** [dpinney/coopforecast#52]
Make model less complex or use Dropout to prevent overfitting

NOTES

2D BASELINE
Epochs: 10, train: 5.46314650492017, Test: 7.257458617545925

2D REMOVE HOLIDAY ENTIRELY
Epochs: 10,  Train: 5.4600476988921915, Test: 6.976594506168758

2D TRAIN ALL YEAR
Epochs: 10,  Train: 6.4945478439331055, Test: 6.5907697677612305

3D BASELINE
Epochs: 10,  Train: 5.143279075622559, Test: 5.208323001861572
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 2: SMALLER
Epochs: 10,  Train: 5.453502655029297, Test: 5.5151472091674805        
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 3: BIGGER
Epochs: 10,  Train: 11.745116233825684, Test: 11.890320777893066
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 4: BIGGER-BUT-SMALLER-ON-END
Epochs: 10,  Train: 6.112486362457275, Test: 6.179825782775879
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 5: BIGGER-ON-END
Epochs: 10,  Train: 4.644232273101807, Test: 4.794312477111816
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 6: SMALLER and MORE_ON_END
Epochs: 10,  Train: 4.366927146911621, Test: 4.4606475830078125
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 7: EXTREME TAIL
Epochs: 10,  Train: 11.97873592376709, Test: 12.041633605957031
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 8: Fix mistake with Test 3
Epochs: 10,  Train: 5.453853607177734, Test: 5.536352157592773
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 9: Fix mistake with Test 4
Epochs: 10,  Train: 4.408908843994141, Test: 4.501700401306152
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

------------------------------------------------------------------------------

BEST
TEST 9: 4.5
TEST 6: 4.46
TEST 5: 4.79

TEST 9
```
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
```

TEST 6
```
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
```

TEST 5
```
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
```

Having a large, fully connected layer after the flattening seems important.

------------------------------------------------------------------------------

TEST 10: Combine 6 and 5
Epochs: 10,  Train: 4.320572853088379, Test: 4.448795318603516
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```

TEST 11: Add another dense layer at tail
Epochs: 10,  Train: 6.968766212463379, Test: 7.061897277832031
```
    layers.Dense(
        ds.train_X.shape[2],
        activation=tf.nn.relu,
        input_shape=(HOURS_AHEAD, ds.train_X.shape[2]),
    ),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2], activation=tf.nn.relu),
    layers.Flatten(),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 2, activation=tf.nn.relu),
    layers.Dense(ds.train_X.shape[2] * HOURS_AHEAD // 4, activation=tf.nn.relu),
    layers.Dense(HOURS_AHEAD),
```
